docs/tutorial/t-create-client-credentials.md

```diff
@@ -0,0 +1,399 @@
+```
+
+#### Trustsore (Server Cert)
+
+We will inject Root CA and Server CA into the truststore file:
+
+```bash
+# ---------- Truststore
+keytool -keystore client.truststore.jks -storepass $KAFKA_CLIENT_TRUSTSTORE_PASSWORD -noprompt \
+  -importcert -alias kafka-ca -file kafka_ca.pem
+keytool -keystore client.truststore.jks -storepass $KAFKA_CLIENT_TRUSTSTORE_PASSWORD -noprompt \
+  -importcert -alias CARoot -file ss_ca.pem
+```
+
+#### Checking certs validity
+
+```bash
+# ---------- Teting Certs validity
+echo "Client certs in Keystore:"
+keytool -list -keystore client.keystore.jks -storepass $KAFKA_CLIENT_KEYSTORE_PASSWORD -rfc | grep "Alias name"
+keytool -list -keystore client.keystore.jks -storepass $KAFKA_CLIENT_KEYSTORE_PASSWORD -v | grep until
+
+echo "Server certs in Truststore:"
+keytool -list -keystore client.truststore.jks -storepass $KAFKA_CLIENT_TRUSTSTORE_PASSWORD -rfc | grep "Alias name"
+keytool -list -keystore client.truststore.jks -storepass $KAFKA_CLIENT_TRUSTSTORE_PASSWORD -v | grep until
+```
+
+### Option A: SASL_SSL
+
+This includes SASL autentication through SSL, which requires:
+
+1. For SSL the client needs to trust the server certificates.
+2. For SASL an username and password credential are required.
+
+To create new users we are going to a SASL_SSL user admin credentials that are stored inside the Kafka server
+
+```bash
+# ---------- Option A: SASL_SSL
+# Create User
+juju ssh kafka/leader "
+echo 'LOG: Creating KAFKA_CLIENT_SASL_USER=$KAFKA_CLIENT_SASL_USER'
+
+sudo charmed-kafka.configs \
+  --bootstrap-server $KAFKA_SERVERS_SASL \
+  --command-config $SNAP_KAFKA_PATH/client.properties \
+  --alter --entity-type=users \
+  --entity-name=$KAFKA_CLIENT_SASL_USER \
+  --add-config=SCRAM-SHA-512=[password=$KAFKA_CLIENT_SASL_PASSWORD]
+"
+```
+
+```bash
+# client-sasl.properties
+echo "sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username=\"safefleet\" password=\"$KAFKA_CLIENT_SASL_PASSWORD\";" > client-sasl.properties
+echo sasl.mechanism=SCRAM-SHA-512 >> client-sasl.properties
+echo security.protocol=SASL_SSL >> client-sasl.properties
+echo bootstrap.servers=$KAFKA_SERVERS_MTLS >> client-sasl.properties
+echo ssl.truststore.location=$SNAP_KAFKA_PATH/client.truststore.jks >> client-sasl.properties
+echo ssl.truststore.password=$KAFKA_CLIENT_TRUSTSTORE_PASSWORD >> client-sasl.properties
+echo ssl.truststore.type=JKS >> client-sasl.properties
+echo ssl.keystore.location=$SNAP_KAFKA_PATH/client.keystore.p12 >> client-sasl.properties
+echo ssl.keystore.password=$KAFKA_CLIENT_KEYSTORE_PASSWORD >> client-sasl.properties
+echo ssl.keystore.type=PKCS12 >> client-sasl.properties
+echo ssl.client.auth=none >> client-sasl.properties
+```
+
+### Option B: mTLS
+
+This is a mutual TLS communication which means:
+
+1. The client needs to trust the server certificates.
+2. Instead of username and passwords, the client needs its own certificate signed by the server for the authentication.
```

Comment:
```suggestion
2. Instead of username and passwords, the client needs its own certificate signed by a certificate trusted by the server for the authentication.
```